src/pages/donate.tsx

```diff
@@ -77,12 +82,40 @@ export function DonatePage() {
         imageFileField = el
     }
 
-    const [imageAdded, setImageAdded] = useState(false)
-
     const handleAddedImage = () => {
-        setImageAdded(true)
+        if (imageFileField.files?.length === 0) {
+            return
+        }
+
+        log("resizing and compressing image")
+        Resizer.imageFileResizer(
+            imageFileField.files?.item(0)!,
+            300,
+            300,
+            "JPEG",
+            90,
+            0,
+            (resized: Blob) => { setImage(resized) },
+            'blob'
+        )
+        log("image resize and compression complete")
```

Comment:
very minor, but should this be inside the callback?

Reply:
It could be, but I was thinking of the callback as more of a next step than the culmination of this.